csrc/scheduler/vectorize_helper.h

```diff
@@ -165,138 +149,29 @@ class TORCH_CUDA_CU_API ProjectedExtent {
 
   std::string toString() const {
     std::stringstream ss;
-    ss << "{";
-    bool first = true;
-    for (auto numer : const_numerator_vals_) {
-      if (!first) {
-        ss << " * ";
-      }
-      first = false;
-
-      ss << numer;
-    }
-
-    for (auto numer : symb_numerator_vals_) {
-      if (numer == nullptr || numer->isOneInt()) {
-        continue;
-      }
-      if (!first) {
-        ss << " * ";
-      }
-      first = false;
-      ss << numer->toInlineString();
-    }
-
-    ss << "} / {";
-    first = true;
-    for (auto denom : const_denominator_vals_) {
-      if (!first) {
-        ss << " * ";
-      }
-      first = false;
-
-      ss << denom;
-    }
-
-    for (auto denom : symb_denominator_vals_) {
-      if (denom == nullptr || denom->isOneInt()) {
-        continue;
-      }
-      if (!first) {
-        ss << " * ";
-      }
-      first = false;
-
-      ss << denom->toInlineString();
-    }
-    ss << "}";
+    ss << numerator()->toInlineString() << " , "
+       << denominator()->toInlineString();
     return ss.str();
   }
 
- private:
-  // Simplify the partial extent as much as possibly by removing common factors
-  // from the numerator and denominator.
-  void simplify() {
-    if (is_simplified_) {
-      return;
-    }
-
-    std::tie(const_numerator_vals_, const_denominator_vals_) =
-        factorization_helpers::removeCommonFactors(
-            const_numerator_vals_, const_denominator_vals_);
-
-    std::tie(symb_numerator_vals_, symb_denominator_vals_) =
-        factorization_helpers::removeSameVals(
-            symb_numerator_vals_, symb_denominator_vals_);
-
-    valid_extents_ = false;
-    is_simplified_ = true;
-  }
-
-  // Simplify the partial extent and generate the Val* for numerator and
-  // denominator by multiplying all the factors.
-  void computeNumerDenomir() {
-    if (valid_extents_) {
-      return;
-    }
-    simplify();
-
-    numerator_ = nullptr;
-
-    int64_t const_numerator_factor = 1;
-    for (auto factor : const_numerator_vals_) {
-      const_numerator_factor *= factor;
-    }
-
-    if (const_numerator_factor > 1) {
-      numerator_ = IrBuilder::create<Int>(const_numerator_factor);
-    }
-
-    for (auto numerator_val : symb_numerator_vals_) {
-      if (numerator_ == nullptr) {
-        numerator_ = numerator_val;
-      } else {
-        numerator_ = SimplifyingIrBuilder::mulExpr(numerator_, numerator_val);
-      }
-    }
-
-    if (numerator_ == nullptr) {
-      numerator_ = isZero() ? FusionGuard::getCurFusion()->zeroVal()
-                            : FusionGuard::getCurFusion()->oneVal();
+  // Run expression simplification and store results in quotient_ and
+  // is_divisible_. This is important because we might use expr evaluator to
+  // evaluate their values multiple times, so we want to simplify them once and
+  // cache the result so that later evaluation can save time.
+  void finalize() {
```

Comment:
In the future, I will assume all extents are positive here. The PR is https://github.com/NVIDIA/Fuser/pull/407. We should think about how to special handle zero extent.